### ISSUE TITLE
invalid union casting in the IGMP module

### ISSUE THREAD

**Issue by mateuszviste** [tass-belgium/picotcp#339]
Since the changes applied by @roel0 in 145464132cefa26220d2baa7e3e1895563799070, the IGMP module won't compile any more using Watcom C. See copied compile-time errors below.

```
wcc -DGENERIC -DPICO_SUPPORT_DHCPC -DPICO_SUPPORT_DNS_CLIENT -DPICO_SUPPORT_ETH -DPICO_SUPPORT_ICMP4 -DPICO_SUPPORT_ICMP6 -DPICO_SUPPORT_IGMP -DPICO_SUPPORT_IPV4 -DPICO_SUPPORT_IPV4FRAG -DPICO_SUPPORT_IPV6 -DPICO_SUPPORT_IPV6FRAG -DPICO_SUPPORT_MCAST -DPICO_SUPPORT_MLD -DPICO_SUPPORT_PING -DPICO_SUPPORT_SNTP_CLIENT -DPICO_SUPPORT_TCP -DPICO_SUPPORT_TFTP -DPICO_SUPPORT_UDP -Iinclude -Imodules -wx -bt=dos -ml -DPICO_COMPILE_TIME=2015 -DDOS_WATCOM modules/pico_igmp.c
Open Watcom C16 Optimizing Compiler Version 1.9
Portions Copyright (c) 1984-2002 Sybase, Inc. All Rights Reserved.
modules/pico_igmp.c(218): Error! E1037: Type cast must be a scalar type
modules/pico_igmp.c(218): Note! I2003: source conversion type is 'struct pico_ip4 '
modules/pico_igmp.c(218): Note! I2004: target conversion type is 'union pico_address '
modules/pico_igmp.c(219): Error! E1037: Type cast must be a scalar type
modules/pico_igmp.c(219): Note! I2003: source conversion type is 'struct pico_ip4 '
modules/pico_igmp.c(219): Note! I2004: target conversion type is 'union pico_address '
modules/pico_igmp.c(499): Error! E1037: Type cast must be a scalar type
modules/pico_igmp.c(499): Note! I2003: source conversion type is 'struct pico_ip4 '
modules/pico_igmp.c(499): Note! I2004: target conversion type is 'union pico_address '
modules/pico_igmp.c(500): Error! E1037: Type cast must be a scalar type
modules/pico_igmp.c(500): Note! I2003: source conversion type is 'struct pico_ip4 '
modules/pico_igmp.c(500): Note! I2004: target conversion type is 'union pico_address '
modules/pico_igmp.c(590): Error! E1037: Type cast must be a scalar type
modules/pico_igmp.c(590): Note! I2003: source conversion type is 'struct pico_ip4 '
modules/pico_igmp.c(590): Note! I2004: target conversion type is 'union pico_address '
modules/pico_igmp.c(591): Error! E1037: Type cast must be a scalar type
modules/pico_igmp.c(591): Note! I2003: source conversion type is 'struct pico_ip4 '
modules/pico_igmp.c(591): Note! I2004: target conversion type is 'union pico_address '
modules/pico_igmp.c: 1156 lines, included 7470, 0 warnings, 6 errors
```

**Comment by roel0** [comment]
fixed in 0c44f7379f652a852484b53d10c1e40aaf1d0469

**Comment by mateuszviste** [comment]
I confirm, the IGMP module compiles all right now. Thank you!
